test/extended/util/disruption/disruption.go

```diff
@@ -366,20 +366,22 @@ func RecordJUnit(f *framework.Framework, name string, fn func() (err error, flak
 	var failure string
 	if err != nil {
 		failure = err.Error()
-
-		if flake {
+	}
+	f.TestSummaries = append(f.TestSummaries, additionalTest{
+		Name:     name,
+		Duration: duration,
+		Failure:  failure,
+	})
+	if flake {
+		if err != nil {
```

Comment:
if we have a flake, why does it matter if err ! =nil?

Reply:
It would show as two passes otherwise as we'd be adding the above testsummary with failure msg of "". I figured we don't want to play around with dupe successes and whatever fun that might cause. Only wanted to add this one if there was a fail.